File: python/ygroorn/traits.py
VIBES = ["playful", "serious", "minimal", "futuristic", "bold"]
THEMES = ["tech", "art", "science", "business", "education"]
LANGUAGES = ["English", "Spanish", "French", "German", "Japanese"]

def get_trait_profile(vibe, theme, language):
    return {
        "vibe": vibe if vibe in VIBES else VIBES[0],
        "theme": theme if theme in THEMES else THEMES[0],
        "language": language if language in LANGUAGES else LANGUAGES[0],
    }

File: ruby/lib/ygroorn/traits.rb
module Ygroorn
  module Traits
    VIBES = %w[playful serious minimal futuristic bold].freeze
    THEMES = %w[tech art science business education].freeze
    LANGUAGES = %w[English Spanish French German Japanese].freeze

    def self.get_trait_profile(vibe, theme, language)
      {
        vibe:     VIBES.include?(vibe) ? vibe : VIBES.first,
        theme:    THEMES.include?(theme) ? theme : THEMES.first,
        language: LANGUAGES.include?(language) ? language : LANGUAGES.first
      }
    end
  end
end